GameServerLib/Extensions.cs

```diff
@@ -28,6 +28,11 @@ public static long ElapsedMicroSeconds(this Stopwatch watch)
             return watch.ElapsedTicks * 1000000 / Stopwatch.Frequency;
         }
 
+        public static float WithHardCaps(this float value, float minValue, float maxValue)
```

Comment:
Normally this is known as a [clamp](https://en.wikipedia.org/wiki/Clamping_(graphics)). Consider naming it `Clamp`.